uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-requests.js

```diff
@@ -9,12 +10,447 @@ define([
+        status_id: prepareStatusFilter(filterByStatus),
+        from_date: {
+          from: filters.selectedPeriod.start_date
+        },
+        to_date: {
+          to: filters.selectedPeriod.end_date
+        },
+        contact_id: prepareContactID()
+      };
+    }
+
+    /**
+     * Returns the contact ID to be used for leave request api
+     *
+     * @return {Object}
+     */
+    function prepareContactID() {
+      if (vm.filters.leaveRequestFilters.selectedUsers) {
+        return vm.filters.leaveRequestFilters.selectedUsers.contact_id;
+      }
+
+      return {
+        "IN": vm.filteredUsers.length ? vm.filteredUsers.map(function (data) {
+            return data.contact_id;
+          }) : ["user_contact_id"]
+      };
+    }
+
+    /**
+     * Returns the status filter to be used for leave request api
+     *
+     * @param {boolean} filterByStatus - if true then leave request api will be filtered using
+     * selected leave request status in the left navigation bar, which would be used to show the
+     * numbers of different status's
+     * @return {Object}
+     */
+    function prepareStatusFilter(filterByStatus) {
+      var filters = vm.filters.leaveRequestFilters,
+        statusFilter = [],
+        //get the value for the waiting_approval status
+        waitingApprovalID = vm.leaveRequestStatuses.find(function (data) {
+          return data.name === 'waiting_approval';
+        }).value;
+
+      //if filterByStatus is true then add the leaveStatus to be used in the leave request api
+      if (filterByStatus && filters.leaveStatus && filters.leaveStatus.value) {
+        statusFilter.push(filters.leaveStatus.value);
+      }
+
+      //if pending_requests is true then add the waiting_approval to be used in the leave request api
+      if (filters.pending_requests && waitingApprovalID) {
+        statusFilter.push(waitingApprovalID);
+      }
+
+      if (statusFilter.length) {
+        return {
+          "IN": statusFilter
+        }
+      }
+    }
+
+    /**
+     * Returns the filter object for contacts api
+     *
+     * @return {Object}
+     */
+    function contactFilters() {
+      var filters = vm.filters.contactFilters;
+
+      return {
+        region: filters.region ? filters.region.value : null,
+        department: filters.department ? filters.department.value : null,
+        level_type: filters.level_type.length ? {
+            "IN": filters.level_type.map(function (data) {
+              return data.value;
+            })
+          } : null,
+        location: filters.location ? filters.location.value : null
+      };
+    }
+
+    /**
+     * Loads the status option values
+     *
+     * @return {Promise}
+     */
+    function loadStatuses() {
+      return OptionGroup.valuesOf('hrleaveandabsences_leave_request_status')
+        .then(function (statuses) {
+          statuses = statuses.concat(allStatus);
```

Comment:
We need the Array to show the statuses in the UI using ng-repeat, so we need an Array . Changed the implementation though and removed the variable.